src/swupd/files.go

```diff
@@ -0,0 +1,143 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+)
+
+type ftype int
+type fmodifier int
+type fstatus int
+
+const (
+	TYPE_UNSET ftype = iota
+	FILE
+	DIRECTORY
+	LINK
+	MANIFEST
+)
+
+const (
+	MODIFIER_UNSET fmodifier = iota
+	CONFIG
+	STATE
+	BOOT
+)
+
+const (
+	STATUS_UNSET fstatus = iota
+	DELETED
+	GHOSTED
+)
+
+// File represents an entry in a manifest
+type File struct {
+	Name    string
+	Hash    string
+	Version uint32
+
+	// flags
+	Type     ftype
+	Status   fstatus
+	Modifier fmodifier
+	Rename   bool
+
+	// renames
+	RenameScore uint16
+	RenamePeer  *File
+
+	Info      os.FileInfo
+	DeltaPeer *File
+}
+
+// setFileTypeFromFlag set file type based on flag byte
+func setFileTypeFromFlag(flag byte, f *File) error {
```

Comment:
Minor comment: I would just make this and other functions below return a value of the respective numeric type -- so they convert from byte to the enum, and then set it at the caller. Still would return error as a second return value.

Reply:
that would be cleaner, thank you.